okhttp/src/main/java/okhttp3/internal/http2/Http2Stream.java

```diff
@@ -316,9 +316,6 @@ synchronized void receiveRstStream(ErrorCode errorCode) {
 
   public synchronized void setHeadersListener(Header.Listener headersListener) {
     this.headersListener = headersListener;
-    if (!headersQueue.isEmpty() && headersListener != null) {
```

Comment:
Whats the context behind this part of the change?  Does it still get accomplished elsewhere? Not needed?

Reply:
I don't actually remember why. Having headers but no listeners can lead the thread to `wait()` so I'd be to revert this change. Tests don't fail either way so I should address that too.
Do you recall @swankjesse ?

Reply:
Reading and setting listeners happen on the same thread on the client side, so there's not point in notifying here because server is not waiting for us anyway.